tools/sgen/Sources/SGen/Generatable/Stylesheet.swift

```diff
@@ -820,271 +820,346 @@ extension Stylesheet {
-        header += "\t\treturn cache.font(name: name, size: size, textStyle: textStyle, weight: weight, traits: traits, traitCollection: traitCollection, isScalable: isScalable)\n"
-        header += "\t}\n\n"
-        
-        header += "\tconvenience init?(name: String, scalable: Bool) {\n"
-        header += "\t\tself.init(name: name, size: 4)\n"
-        header += "\t\tself.isScalable = scalable\n"
-        header += "\t}\n\n"
-        
-        header += "\tvar isScalable: Bool {\n"
-        header += "\t\tget { return objc_getAssociatedObject(self, &__ScalableHandle) as? Bool ?? false }\n"
-        header += "\t\tset { objc_setAssociatedObject(self, &__ScalableHandle, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC) }\n"
-        header += "\t}\n\n"
-        
-        header += "\tvar textStyle: UIFont.TextStyle? {\n"
-        header += "\t\treturn fontDescriptor.fontAttributes[.textStyle] as? UIFont.TextStyle\n"
-        header += "\t}\n\n"
-
-        header += "\tvar fixedFont: UIFont {\n"
-        header += "\t\tif isScalable == false { return self }\n"
-        header += "\t\tif let fontType = fontType {\n"
-        header += "\t\t\treturn UIFont.font(name: fontType.name, size: fontType.size, textStyle: fontType.textStyle, weight: fontType.weight, traits: fontType.traits, traitCollection: fontType.traitCollection, isScalable: false)\n"
-        header += "\t\t}\n"
-        header += "\t\tguard let textStyle = textStyle, let defaultSize = defaultSizes[textStyle] else { return self }\n"
-        header += "\t\tlet fontDescriptor = UIFontDescriptor.preferredFontDescriptor(withTextStyle: textStyle)\n"
-        header += "\t\treturn UIFont(descriptor: fontDescriptor, size: defaultSize)\n"
-        header += "\t}\n\n"
-        
-        header += "}\n\n"
-        
+
+        header += """
+    func with(traits: UIFontDescriptor.SymbolicTraits) -> UIFont {
+        let descriptor = fontDescriptor.withSymbolicTraits(traits)
+        return UIFont(descriptor: descriptor!, size: 0)
+    }
+
+    class func font(name: String? = nil,
+                    size: CGFloat? = nil,
+                    textStyle: \(textStyleType)? = nil,
+                    weight: UIFont.Weight? = nil,
+                    traits: UIFontDescriptor.SymbolicTraits,
+                    traitCollection: UITraitCollection? = nil,
+                    isScalable: Bool = true) -> UIFont {
+        return cache.font(name: name,
+                          size: size,
+                          textStyle: textStyle,
+                          weight: weight,
+                          traits: traits,
+                          traitCollection: traitCollection,
+                          isScalable: isScalable)
+    }
+
+    convenience init?(name: String, scalable: Bool) {
+        self.init(name: name, size: 4)
```

Comment:
why 4?

Reply:
No idea where this number came from, but as we spoke offline, we're removing this as it's not needed.